File: app/src/main/java/tech/destinum/machines/data/POJO/Machine.java
package tech.destinum.machines.data.POJO;

import android.arch.persistence.room.Embedded;
import android.arch.persistence.room.Entity;
import android.arch.persistence.room.ForeignKey;
import android.arch.persistence.room.PrimaryKey;

import static android.arch.persistence.room.ForeignKey.CASCADE;

@Entity(tableName = "machines")
public class Machine {

    @PrimaryKey(autoGenerate = true)
    private long id;
    private String name;

    @Embedded
    private Income income;

    public Income getIncome() {
        return income;
    }

    public void setIncome(Income income) {
        this.income = income;
    }

    public Machine(String name) {
        this.id = id;
        this.name = name;
    }

    public long getId(){
        return id;
    }

    public void setId(long id){
        this.id = id;
    }

    public String getName(){
        return name;
    }

    public void setName(String name){
        this.name = name;
    }

}

File: app/src/main/java/tech/destinum/machines/Data/Local/Entities/Machine.kt
package tech.destinum.machines.Data.Local.Entities

import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey


@Entity(tableName = "machines")
class Machine {

    @PrimaryKey(autoGenerate = true)
    var id: Long = 0
    var name: String? = null
    var total_income: Double = 0.toDouble()

    @Ignore
    var total_amount: String? = null

    constructor(name: String, total_income: Double) {
        this.name = name
        this.total_income = total_income
    }

    @Ignore
    constructor(name: String) {
        this.name = name
    }

}
